### ISSUE TITLE
False Positive: found duplicate lang item panic_impl

### ISSUE THREAD

**Issue by vikigenius** [rust-lang/rust-analyzer#10659]
I am following along this blog post: https://os.phil-opp.com/freestanding-rust-binary/

This is the MRC example I have.

```rust
#![no_std]
#![no_main]
use core::panic::PanicInfo;

/// This function is called on panic.
#[panic_handler]
// Rust Analyzer marks this line as an error found duplicate lang item panic_impl
fn panic(_info: &PanicInfo) -> ! {  
    loop {}
}


#[no_mangle]
pub extern "C" fn _start() -> ! {
    loop {}
}
```

The project builds fine after adding target

`rustup target add thumbv7em-none-eabihf`

and building with `cargo build --target thumbv7em-none-eabihf`

Rust analyzer should not mark the panic as an error.
